core/commands/storage.go

```diff
@@ -265,7 +292,7 @@ func retryMonitor(ctx context.Context, ss *storage.Session, n *core.IpfsNode, p
 		go func(chunkHash string, chunkInfo *storage.Chunk) {
 			candidateHost, err := getValidHost(retryQueue)
 			if err != nil {
-				ss.SetStatus(errStatus)
+				sendSessionStatusChan(ss.TimeOutChan, storage.InitStatus, false, err)
```

Comment:
TimeOutChan may not be the best name here - here you are just returning an error, it's not from a timeout